pandas/core/arrays/arrow/dtype.py

```diff
@@ -148,6 +148,14 @@ def name(self) -> str:  # type: ignore[override]
     @cache_readonly
     def numpy_dtype(self) -> np.dtype:
         """Return an instance of the related numpy dtype"""
+        if pa.types.is_timestamp(self.pyarrow_dtype):
```

Comment:
Could you also raise an issue on the Arrow issue tracker to suggest they not hardcode the resolution? Then it would be good to reference that issue here noting this logic can be removed if/when they address it

Reply:
yep, opened https://github.com/apache/arrow/issues/34462 and added comments